src/merkle/mod.rs

```diff
@@ -15,6 +15,12 @@ use crate::Value;
 use hashes::{sha256, Hash, HashEngine};
 use std::fmt;
 
+/// Trait for types that have a Commitment Merkle Root.
+pub trait Merkelized {
```

Comment:
In e54c81c286f773a8f1b8727a9fe96e168ccb52d5:

I would call this trait `Cmr`...